web-console/src/views/workbench-view/run-panel/run-panel.tsx

```diff
@@ -485,7 +511,10 @@ export const RunPanel = React.memo(function RunPanel(props: RunPanelProps) {
             }
           >
             <Button
-              text={`Engine: ${queryEngine || `auto (${effectiveEngine})`}`}
+              text={`Engine: ${
+                (enginesLabelFn ? enginesLabelFn(queryEngine).text : queryEngine) ||
+                `auto (${enginesLabelFn ? enginesLabelFn(effectiveEngine) : effectiveEngine})`
```

Comment:
should this be `enginesLabelFn(effectiveEngine).text` ?

Reply:
I think a better structure would be instead of letting `enginesLabelFn` possibly be undefined throughout the whole component and having to check every time you use it to do `const enginesLabelFn = props.enginesLabelFn || DEFAULT_ENGINES_LABEL_FN;` at the top of the component and then implement everything with `enginesLabelFn` always being called.

Ditto for `maxTaskLabelFn`